### PR TITLE
Issue5 userprofileput

### PR DESCRIPTION
**Pull request by bschuste**
@bretwalker @TechGladiator @mattpeyton

### CODE REVIEW

File: api/models/user.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const Sequelize = require('sequelize');
+
+module.exports = function(sequelize, DataTypes){
+    return sequelize.define('users', {
```
Comment (bretwalker) [inline api/models/user.js]: You'll want to add columns to match this table: https://github.com/codelouisvilleproject/database-schema/blob/master/sql/V2__AddUsersAndActivities.sql#L3

We'll also need a PR on `database-schema` to add an `email` column.

File: api/swagger/swagger.yaml [hunk 2]
```diff
@@ -62,8 +62,64 @@ paths:
             $ref: "#/definitions/ErrorResponse"
   /swagger:
     x-swagger-pipe: swagger_raw
+  /users/{userId}:
```
Comment (bretwalker) [inline api/swagger/swagger.yaml:65]: 👍

File: api/controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  var birthDay = req.swagger.params.userInfo.xbirthDay || 1;
+  var birthYear = req.swagger.params.userInfo.birthYear || 1970;
+  var email = req.swagger.params.userInfo.email || 'john.doe@users.com';
+  var response = util.format('User ID:%d,', id);
+  response += util.format('firstName:%s,', firstName);
+  response += util.format('lastName:%s,', lastName);
+  response += util.format('birthDate:%d-%d-%d,', birthMonth, birthDay, birthYear);
+  response += util.format('email:%s', email);
+
+  res.json({'message': response});
```
Comment (bretwalker) [inline api/controllers/userController.js]: This is fine for now, but we'll probably want to just return a `200` with a simple message like `OK`.
Once we add in validation, we can get rid of the default (`John Doe`) values.

File: api/db.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 const Sequelize = require('sequelize');
-const CodeLouisvilleStudents = require('./models/codeLouisvilleStudents')
```
Comment (bretwalker) [inline api/db.js:2]: You can delete the file too.

**Review by bretwalker — APPROVED** [review]
A couple of comments, but overall, it looks good.
Are you planning on adding in the `Sequelize` logic before you merge, or were you just wanting to get this merged so others can use it as a starting point?

File: api/controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  var id = req.swagger.params.userId.value;
+  var firstName = req.swagger.params.userInfo.value.firstName;
+  var lastName = req.swagger.params.userInfo.value.lastName;
+  var birthMonth = req.swagger.params.userInfo.value.birthMonth;
+  var birthDay = req.swagger.params.userInfo.value.birthDay;
+  var birthYear = req.swagger.params.userInfo.value.birthYear;
+  var birthDate = util.format('%d-%d-%d,', birthYear, birthMonth, birthDay);
+  var email = req.swagger.params.userInfo.value.email;
+
+  if ((birthDay>0 && birthDay<32) && (birthMonth>0 && birthMonth<13) && birthYear) {
```
Comment (bretwalker) [inline api/controllers/userController.js:20]: `if !isNaN(Date.parse(birthDate))`

File: api/swagger/swagger.yaml [hunk 6]
```diff
@@ -62,8 +62,61 @@ paths:
...
+        type: "string"
+      birthMonth:
+        type: "integer"
+      birthDay:
+        type: "integer"
+      birthYear:
+        type: "integer"
+      email:
+        type: "string"
+  UsersResponse:
```
Comment (bretwalker) [inline api/swagger/swagger.yaml:114]: We don't need this for this story.

**Review by bretwalker — APPROVED** [review]
There are two possible changes you could make, but they're for code cleanliness, not functionality. Feel free to merge. Good work!

File: api/swagger/swagger.yaml (hunk 6 above)
Comment (bschuste) [reply api/swagger/swagger.yaml:114]: Do you mean UsersResponse?

File: api/controllers/userController.js (hunk 5 above)
Comment (bschuste) [reply api/controllers/userController.js:20]: Tested with birthYear to 0 and did not reply 400 but 200.

**Pull request merged**